Source/Parser.swift

```diff
@@ -182,25 +149,56 @@ public class Parser {
 
         - returns: The result Dictionary.
     */
-    public class func parseProperties(json: AnyObject, closure: (ParserPropertyMaker) -> Void) throws -> [String: Any] {
-        if let json = json as? [String: AnyObject] {
-            return try parsePropertiesForJSONDictionary(json, closure: closure)
+    public class func parseProperties(
+        from json: AnyObject,
+        using closure: (ParserPropertyMaker) -> Void)
+        throws -> [String: Any]
+    {
+        if let data = json as? Data {
+            let result: [String: Any]
+
+            do {
+                let options = JSONSerialization.ReadingOptions(rawValue: 0)
+                let json = try JSONSerialization.jsonObject(with: data, options: options)
+
+                result = try parseProperties(from: json, using: closure)
+            } catch {
+                if error is ParserError {
+                    throw error
+                } else {
+                    let error = error as NSError
+
+                    throw ParserError.deserialization(
+                        failureReason: "JSON data deserialization failed with error: \"\(error.description)\""
+                    )
+                }
+            }
+
+            return result
```

Comment:
These were intentionally separate APIs in the past to add clarity and strong typing for the `Data` option being available. The design guidelines state that weak type information should be compensated for with a noun to describe its role. In this case we may consider splitting these back up into `parseProperties(fromJSON: ...` and `parseProperties(fromData: ...`. Thoughts?